src/app/frontend/deploy/kdfilereader_directive.js

```diff
@@ -0,0 +1,49 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Provides ng-model binding for input elements with file type and updates the model
+ * on change with {name:string, content:string} custom object form of the selected file.
+ *
+ * @param {!angular.$log} $log
+ * @return {!angular.Directive}
+ * @ngInject
+ */
+export default function kdFileReaderDirective($log) {
+  return {
+    restrict: 'A',
+    require: ['^kdUpload', 'ngModel'],
+    link: function(scope, element, attrs, ctrls) {
+      let kdUploadController = ctrls[0];
```

Comment:
@type {KdUploadController} is unknown. If I import it then I need to use it otherwise I get another style error. Therefore I haven't set the type yet. Do you have any suggestion?